src/daemon/messages.go

```diff
@@ -221,30 +223,47 @@ func (gpm *GivePeersMessage) Process(d Daemoner) {
 	d.AddPeers(peers)
 }
 
-// IntroductionMessage jan IntroductionMessage is sent on first connect by both parties
+// IntroductionMessage is sent on first connect by both parties
 type IntroductionMessage struct {
-	// Mirror is a random value generated on client startup that is used
-	// to identify self-connections
+	// Mirror is a random value generated on client startup that is used to identify self-connections
 	Mirror uint32
 	// Port is the port that this client is listening on
 	Port uint16
 	// Protocol version
 	Version int32
 	c       *gnet.MessageContext `enc:"-"`
 	// We validate the message in Handle() and cache the result for Process()
-	valid bool `enc:"-"` // skip it during encoding
+	valid         bool           `enc:"-"` // skip it during encoding
+	userAgentData useragent.Data `enc:"-"`
```

Comment:
Could it be possible to limit user agent ID to be of a fixed number of chars (e.g. similar to message prefix) ?